odoo/api.py

```diff
@@ -632,25 +632,79 @@ def __call__(self, cr=None, user=None, context=None, su=None):
+
+        :param str xml_ids: record xml_ids, under the format ``<module>.<record_id>``
+        :param bool raise_if_not_found: whether the method should raise if any ``xml_id`` isn't found.
+        :returns: Found records or None
+        :raise ValueError: if any record wasn't and ``raise_if_not_found`` is True,
+            or if the user mixed ``xml_ids`` that belong to different models.
+
+        .. note::
+            If ``raise_if_not_found`` is ``False`` and there are multiple
+            ``xml_ids`` as input, then it won't be possible to determine which
+            specific records weren't found by only looking at the recordset.
+
+        .. warning::
+            | ``raise_if_not_found`` is now a keyword-only argument.
+            | Its use as a non-keyword argument is **deprecated** but it's still working.
+
+            For example::
+
+                self.env.ref('base.us', 'base.x', False)
+                DeprecationWarning: Since 19.0, raise_if_not_found is a keyword-only argument
+                res.country(233)
+
+                self.env.ref('base.us', 'base.x', raise_if_not_found=False)
+                res.country(233)
+
+                self.env.ref('base.us', 'base.x')
+                ValueError: External ID not found in the system: base.x
+
+                self.env.ref('base.us', 'base.x', True)
+                DeprecationWarning: Since 19.0, raise_if_not_found is a keyword-only argument
+                ValueError: External ID not found in the system: base.x
+
+                self.env.ref('base.us', 'base.x', raise_if_not_found=True)
+                ValueError: External ID not found in the system: base.x
 
-        :param str xml_id: record xml_id, under the format ``<module.id>``
-        :param bool raise_if_not_found: whether the method should raise if record is not found
-        :returns: Found record or None
-        :raise ValueError: if record wasn't found and ``raise_if_not_found`` is True
         """
-        res_model, res_id = self['ir.model.data']._xmlid_to_res_model_res_id(
-            xml_id, raise_if_not_found=raise_if_not_found
-        )
 
-        if res_model and res_id:
-            record = self[res_model].browse(res_id)
-            if record.exists():
-                return record
-            if raise_if_not_found:
-                raise ValueError('No record found for unique ID %s. It may have been deleted.' % (xml_id))
-        return None
+        # Avoid breaking the API when raise_if_not_found was given as a pure boolean non-keyword arg
+        if len(xml_ids) > 1 and raise_if_not_found is True and isinstance(xml_ids[-1], bool):
+            warnings.warn("Since 19.0, raise_if_not_found is a keyword-only argument", DeprecationWarning)
+            raise_if_not_found = xml_ids[-1]
+            xml_ids = xml_ids[:-1]
+
+        id_xmlid = {}
+        res_ids = []
+        first_model_name, first_xml_id = None, None
+        for xml_id in xml_ids:
```

Comment:
Sorry if i wrong, is the `xml_ids` is a string separated by comma right ? so here it should be something like xml_ids`.split(",")` maybe